### DISCUSSION TITLE
Save as new asset should select the newly save asset from inside entry editor

### DISCUSSION THREAD

**Discussion by RitterKnightCreative** [craftcms/cms#8547] [Ideas]
### Description

When editing an image inside Craft, it's easy to go into the Edit window to quickly fix a crop or rotation problem with an image.

In the context of editing an entry, however, unless you choose "Save" (overwriting the original image), you probably want to Save as a New Asset. So having that as the rightmost (dare I say "default") button makes sense. 

However, Craft doesn't go ahead and *also* choose that newly edited image for you in the editing window, you end up having to take that step yourself manually. 

It would be a much smoother workflow if the happened.

(FWIW if the "Save as new Asset" button weren't red and also the default choice, I think that it'd be a more obvious sell.)

### Steps to reproduce

1. Select an image asset when editing an entry
2. Choose Edit to edit the image
3. Make changes to said asset
4. Choose Save as new asset to close the window
5. Remove (x) the image from the editor.
6. Tap "choose" button to select newly edited image.

### What should happen instead
Craft should simply use the newly edited image, skipping Steps 5 and 6.

Alternatively, if you want to keep the current behavior,  maybe slap an unchecked checkbox in there to "Use new image" which would only appear in the edit entry window. Ticking the would do the same thing, eliminating steps 5 and 6.

**Comment by davidhellmann** [comment]
@brandonkelly I have a similar problem.

We crop an Image and click `save as new asset`
The workflow from @RitterKnightCreative would be much much better but my problem here is. After I click save as new asset => remove the currently selected image => I can't find the new image. I've to do a reload and then it works.

Looks like a bug for me?

Craft Version: 3.0.36

**Comment by dennisfrank** [comment]
I had the same issue as @davidhellmann with a client today. 

1. She expected that “Save as a New Asset” would update the asset field with the newly created asset.
2. Even after removing the original asset from the field the new asset could not be found without a reload.

I hope this could be fixed because right now it’s very confusing for technically less savvy users.

**Comment by engram-design** [comment]
+1 on this, I know its a bit technical, but it makes the most sense.

**Comment by melbazany** [comment]
Echoing @dennisfrank and @davidhellmann, but in Craft 5.6. Could this be added as config option to modify the default behavior? We've run into a handful of instances where clients are expecting the newly created asset to be the chosen file, not the original asset.
